src/parser.rs

```diff
@@ -60,36 +60,60 @@ pub fn parse_term(sig: &mut Signature, input: &str) -> Result<Term, ParseError>
 ///
 /// # TRS syntax
 ///
-/// `input` is parsed as a `<program>`, defined as follows:
+/// The input is parsed as a `program`, defined as follows in [augmented Backus-Naur form]:
 ///
 /// ```text
-/// <program> ::= ( <comment>* <statement> ";" <comment>* )*
+/// program = *wsp *( *comment statement ";" *comment ) *wsp
 ///
-/// <comment> ::= "#" <any-char-but-newline> "\n"
+/// statement = rule / top-level-term
 ///
-/// <statement> ::= <rule> | <top-level-term>
+/// rule = top-level-term *wsp "=" *wsp top-level-term
+/// rule /= rule *wsp "|" *wsp top-level-term
 ///
-/// <rule> ::= <top-level-term> "=" <top-level-term> ( "|" <top-level-term> )
+/// top-level-term = term / "(" top-level-term ")"
+/// top-level-term /= top-level-term 1*wsp top-level-term
 ///
-/// <top-level-term) ::= ( <term> | ( "(" <top-level-term> ")" ) ) (" "  ( <term> | ( "(" <top-level-term> ")" ) ) )*
+/// term = variable / application
 ///
-/// <term> ::= <variable> | <application>
+/// variable = identifier"_"
 ///
-/// <variable> ::= <identifier>"_"
+/// application = identifier "(" *( *wsp term *wsp ) ")"
```

Comment:
There *must* be whitespace between subsequent terms in an application, so this definition needs to revised, I think.

Reply:
It actually isn't required by the parser. We could though — and probably should. I'll add that.